api/pkg/db/migration/202107221500_add_ondelete_constraints.go

```diff
@@ -0,0 +1,146 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+// Drops existing foreign key constraints and create the same constraints
+// in order to add another OnDelete constraints to foriegn key
+func addOnDeleteConstraints(log *log.Logger) *gormigrate.Migration {
+
+	return &gormigrate.Migration{
+		ID: "202107221500_add_ondelete_constraints",
+		Migrate: func(db *gorm.DB) error {
+			txn := db.Begin()
+
```

Comment:
Can you create a function like this https://github.com/tektoncd/hub/blob/main/api/pkg/db/migration/202107291608_refresh_all_tables.go#L36
and handle the error in a different function instead of writing `txn.Rollback` multiple times